sympy/functions/elementary/hyperbolic.py

```diff
@@ -472,6 +425,46 @@ def _eval_is_real(self):
         re, im = arg.as_real_imag()
         return (im%pi).is_zero
 
+    def _eval_is_positive(self):
+        # cosh(x+I*y) = cos(y)*cosh(x) + I*sin(y)*sinh(x)
+        # cosh(z) is positive iff it is real and the real part is positive.
+        # So we need sin(y)*sinh(x) = 0 which gives x=0 or y=n*pi
+        z = self.args[0]
+
+        x, y = z.as_real_imag()
+        ymod = y % (2*pi)
+
+        # Case 1 (y=n*pi): cosh(z) = (-1)**n * cosh(x) -> positive for n even
+        def case_y_npi():
+            return ymod.is_zero
+
+        # Case 2 (x=0): cosh(z) = cos(y) -> positive when cos(y) is positive
+        def case_x_zero_and():
+            yield x.is_zero
+            yield fuzzy_or([fuzzy_bool(ymod < pi/2), fuzzy_bool(ymod > 3*pi/2)])
+
+        def cases():
+            yield case_y_npi()
+            yield fuzzy_and(case_x_zero_and())
+
+        return fuzzy_or(cases())
+
+    def _eval_is_nonnegative(self):
+        z = self.args[0]
```

Comment:
This is the simplified version that I was describing. Since we can say the conditions for cosh(z) being zero with if and only if we can simply return a statement built up with fuzzy logic operators.